### ISSUE TITLE
Introduce the b-nav-item-toggle component that works with v-b-toggle

### ISSUE THREAD

**Issue by thebspin** [bootstrap-vue/bootstrap-vue#5391]
### Is your feature request related to a problem? Please describe...

Working with a vertical `b-nav` component i want to have options that are hidden by a toggle.
I found the `v-b-toggle` directive but that does not work with the `b-nav-item` component. 
You can work around that using the `b-nav-text` component instead. That however comes without the styling of a nav item.
I could also go for a v-model solution or a method based solution but that's more lines of code, less flexible and defeats the purpose of the directive itself.

I understand this is because the `b-nav-item` component works with `evt.preventDefault()`.

### Describe the solution you'd like

If there would be an `b-nav-item-toggle` component that would be expected to be used with the directive, then you would no longer need the `preventDefault()` check and i could do something like:

```
<b-nav-item-toggle v-b-toggle.secret>Dropdown</b-nav-item>
<b-collapse id="secret">
    <b-nav-item>Do you see me?</b-nav-item>
</b-collapse>
```

And there would be no need for v-model and thus data opbjects or a method.

### Describe alternatives you've considered

An alternative could be that the `b-nav-text` component could be styled to look like the  `b-nav-item` 

I think vertical navbar's would greatly benefit from this (and thusfor makes it easier to make admin themes, something i have not found using this otherwise great framework)

**Comment by thebspin** [comment]
If i uderstand this correctly it will no longer check for the `defaultPrevend` event right? Having a dedicated component would allow for more flexibility. Lets say i have an vertical nav that shows a chevron when something is toggleable. This chevron then would change once something is toggeled open or closed. My understanding is that this is only possible right now with a lot of custom code.

**Comment by tmorehouse** [comment]
`<b-nav-item.` is based on b-link, which calls `evt.preventDefault()` if the href is `#` (to prevent scroll to top behaviour). PR #5396 changes v-b-toggle so that it doesn't check for `defaultPrevented`, which will make `v-b-toggle` for on `<b-nav-item>` that doesn't' have a link )i..e href is not set)

**Comment by thebspin** [comment]
I see, so if i want to keep the toggle state for example i could so something like:
```
toggle(e) {
      e.target.classList.toggle("toggled")
}
```

Because there is no other method implemented for this in the framework?

**Comment by tmorehouse** [comment]
`v-b-toggle` controls the `collapsed` and `not-collapsed` classes on the element that is the trigger for the collapse (or sidebar).  You can use these classes to style the content in your toggle element.

**Comment by thebspin** [comment]
Ah that is nice! Can't wait for the next release then, thanks for the fast action you took! :)

**Comment by tmorehouse** [comment]
The classes are available in 2.14.0, but there are issues with default href of `#`.  Setting the `b-nav-item` `href` to something other than `#` (the default) will get around this issue (i.e. set prop `href` to `#{id-of-collapse}` is a temporary fix).
